### PR TITLE
Add congruence.periodic_union.prefix_count.compute operation

### LINKED ISSUE #2798: [Number theory] Count finite periodic congruence unions through an exact cutoff
## Parent gap / RFC

Erdős #488 is live-open with zero claimed proofs.  For a finite divisor family
`A`, its set of multiples is the periodic congruence union

```text
B = union_{a in A} { x : x = 0 (mod a) }.
```

The problem compares exact prefix counts `|B ∩ [1,n]|` and `|B ∩ [1,m]|`.
Jacobian already computes the density/count over one common period for a
`PeriodicCongruenceUnionSource`, but it cannot return the exact count through
an arbitrary integer cutoff.  The latter is a general periodic-set primitive,
also useful for residue-sieve experiments and finite density comparisons.

## Public mathematical postcondition

Given an admitted finite periodic congruence union/complement and a nonnegative
integer cutoff `X`, return exactly the number of integers in `[1,X]` that
belong to the declared periodic set.

## Proposed operation ID

`congruence.periodic_union.prefix_count.compute`

## Input

The existing canonical `PeriodicCongruenceUnionSource` (finite union of
residue subsets, optionally complemented) and one nonnegative canonical
integer `cutoff`.  It must reuse the source's exact normalization and common
period semantics rather than introduce a divisor-family-only duplicate type.

## Result

A source-bound `PeriodicCongruenceUnionPrefixCountResult` containing the
normalized source, its common period, the cutoff, and the exact count in
`[1, cutoff]`.  The interval convention is explicit: zero contributes no
integer, while a residue class zero contributes its positive multiples.

## Completeness semantics

Every admitted request returns one exact count.  No asymptotic density,
limsup, or extremal inequality is inferred.  Requests above input/intermediate
or result-digit bounds are rejected before computation; an absent proof of a
large-period prefix must never become a count.

## Bounds and cost class

Let `L` be the source's least common period, `q,r = divmod(X,L)`, and `c` the
one-period occupied count.  The exact answer is

```text
q*c + |S ∩ [1,r]|,
```

where `S` is the canonical occupied residue set in one period with the zero
residue interpreted as `L` for positive-integer counting.  Admission must
reuse/prove the source's common-period and residue-lifting work, bound any
needed prefix materialization/lookup through `L`, and separately bound cutoff
and result digit lengths.  A huge `X` can remain admissible when `L` and the
result are small; no scan through `[1,X]` is permitted.

## Execution ownership

- Request-bounds owner and execution-plan quantities: number-theory periodic
  domain; source rows, common period, materialized/queried prefix work,
  cutoff digits, quotient-product digits, and result bytes.
- Maintained backend or Jacobian kernel adapter: extend the existing exact
  periodic-union kernel; no enumeration of the requested prefix and no solver.
- Canonical result construction and malformed-backend handling: preserve the
  normalized source and exact source/period/cutoff/count binding.
- Independent-result verifier: none; the public API computes the value.
- Serialized-result bound and transport projection: a constant-size scalar
  result plus the existing normalized source.
- Native/MCP semantic parity and transport-only differences: identical count.

## Acceleration and exact representation

Compute one common-period occupancy structure using the existing bounded
periodic-union engine, then use Euclidean division and a prefix count.  A later
count-only inclusion--exclusion or compressed-residue kernel may replace the
implementation only if it preserves the same result and admitted source
semantics.  It must not change the public operation into a density estimator.

## Maintained implementation

This is a thin domain-owned extension of the existing periodic-union kernel;
its arithmetic is exact integers and canonical residue sets.  It is more
proportionate than a generic divisibility-union DSL and more general than a
#488-specific multiple-count function.

## Defining invariant

The returned count equals the cardinality of

```text
{ t in {1,...,X} : exists source subset (m,R) with t mod m in R }
```

(or its declared complement).  On one full period it agrees with the existing
occupied count; splitting `[1,X]` into `q` full periods and one prefix gives
the stated reconstruction identity.

## Evidence plan

Test empty/full unions; residue zero; one nonzero residue; cutoff zero; exact
period endpoints; `X=qL+r` across several periods; complements; overlapping
and nested moduli; negative/oversized input residues normalized by the shared
source; and exhaustive direct enumeration on small periods/cutoffs.  Test the
multiples encoding of a divisor family as a #488-shaped fixture, comparing
cross-multiplied ratios only in the test/caller—not in the operation.  Add
cutoff-digit and period/materialization boundary cases.

## Why this is not caller-side composition

`congruence.periodic_union.measure.compute` gives only one-period cardinality
and rational density; `profile.compute` emits representatives only when full
materialization fits.  The source-bound quotient/remainder decomposition,
positive-integer treatment of residue zero, prefix coverage, large-cutoff
integer arithmetic, and exact count are otherwise caller-owned.  They form a
stable periodic-set postcondition independent of #488.

## Deferred operation family

Do not add the #488 ratio decision, an optimiser over divisor families,
a counterexample search, an asymptotic density claim, automatic eventual
periodicity recognition, or arbitrary Boolean formulas.  Those have different
mathematical and search semantics.

### PR DESCRIPTION
**Pull request by morluto**
## Summary

Implements `congruence.periodic_union.prefix_count.compute` from issue #2798.

Given a finite periodic congruence union (optionally complemented) and a nonnegative integer cutoff, returns the exact number of integers in [1, cutoff] that belong to the declared periodic set.

## Design

- **Operation ID**: `congruence.periodic_union.prefix_count.compute`
- **Input**: `PeriodicCongruenceUnionSource` +  (CanonicalInteger)
- **Output**: `PeriodicUnionPrefixCountResult` with exact count, common period, and occupied count
- **Kernel**: uses the existing  to materialize one period, then applies the periodicity formula:  where 
- **Bounds**: reuses existing periodic source normalization and common-period infrastructure

## Tests

- Fixture: union of 0 mod 2 and 1 mod 3 on [1,6] has count 4
- Empty union: count 0
- 0 mod 1: all integers (count = cutoff)
- Cutoff 0: count 0
- Periodicity: extending cutoff by one period increases count by occupied_count
- Source preservation

Closes #2798

[Continue this on Linzumi](https://serve.linzumi.com/w/williamlin1327-workspace/thread/general/75e7770a-f231-4c70-a9d6-4f60aa64c75a)

### CODE REVIEW

File: src/jacobian/math/number_theory/periodic_prefix_count/_models.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class PeriodicUnionPrefixCountResult(StrictModel):
+    """The exact count of integers in [1, cutoff] belonging to the periodic set."""
+
+    source: PeriodicCongruenceUnionSource
+    cutoff: CanonicalInteger
+    common_period: CanonicalInteger
+    occupied_count: CanonicalInteger
+    count: CanonicalInteger
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/_models.py]: Fixed on the PR branch with the corresponding owner-admission or boundedness change; focused owner tests and Ruff validation pass.

File: src/jacobian/math/number_theory/periodic_prefix_count/operations.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+__all__ = ["compute_periodic_union_prefix_count"]
+
+MAX_RESULT_BYTES = CanonicalLimits().max_output_bytes
+
+
+def _reject(code: str, message: str) -> NoReturn:
+    raise OperationDomainValidationError(
+        location=("source",),
+        code=f"number_theory.periodic_prefix.{code}",
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/operations.py]: Fixed on the PR branch with the corresponding owner-admission or boundedness change; focused owner tests and Ruff validation pass.

File: src/jacobian/math/number_theory/periodic_prefix_count/_models.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class PeriodicUnionPrefixCountResult(StrictModel):
+    """The exact count of integers in [1, cutoff] belonging to the periodic set."""
+
+    source: PeriodicCongruenceUnionSource
+    cutoff: PeriodicPrefixCutoff
+    common_period: PeriodicPositiveInteger
+    occupied_count: PeriodicNonnegativeInteger
+    count: PeriodicNonnegativeInteger
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/_models.py:51]: Fixed on the PR branch with the corresponding owner-admission or boundedness change; focused owner tests and Ruff validation pass.

File: src/jacobian/math/number_theory/periodic_prefix_count/_models.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    @model_validator(mode="after")
+    def require_count_invariants(self) -> Self:
+        cutoff = int(self.cutoff)
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/_models.py]: Fixed on the PR branch with the corresponding owner-admission or boundedness change; focused owner tests and Ruff validation pass.

File: src/jacobian/math/number_theory/periodic_prefix_count/_models.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+from jacobian.math.number_theory._periodic_models import (
+    PeriodicCongruenceUnionSource,
+)
+
+
+class PeriodicUnionPrefixCountRequest(StrictModel):
+    """Request for the prefix count of a periodic congruence union."""
+
+    source: PeriodicCongruenceUnionSource
+    cutoff: CanonicalInteger
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/_models.py]: Fixed on the current branch. Negative cutoffs are rejected by the owner-local native admission before kernel execution, with the same nonnegative domain enforced for catalog calls.

File: src/jacobian/math/number_theory/periodic_prefix_count/_models.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class PeriodicUnionPrefixCountResult(StrictModel):
+    """The exact count of integers in [1, cutoff] belonging to the periodic set."""
+
+    source: PeriodicCongruenceUnionSource
+    cutoff: CanonicalInteger
+    common_period: CanonicalInteger
+    occupied_count: int
+    count: int
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/_models.py]: Fixed on the current branch. Prefix counts are represented as canonical decimal strings in the result, so values above JSON’s safe integer range serialize exactly.

File: src/jacobian/math/number_theory/periodic_prefix_count/operations.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        )
+
+    q, r = divmod(cutoff, period)
+    count = q * occupied_count
+    for res in range(1, r + 1):
+        if res in occupied:
+            count += 1
+    return PeriodicUnionPrefixCountResult(
+        source=source,
+        cutoff=str(cutoff),
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/operations.py]: Fixed on the current branch. The operation preserves the validated canonical cutoff text and formats the native integer with format_canonical_integer; it no longer routes large values through str(int).

File: src/jacobian/math/number_theory/periodic_prefix_count/operations.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    source: PeriodicCongruenceUnionSource,
+    cutoff: int,
+) -> PeriodicUnionPrefixCountResult:
+    """Return the exact count of integers in [1, cutoff] belonging to the periodic set.
+
+    Uses the periodicity: if the common period is L and c residues are
+    occupied in one period, then the count through X is q*c + r,
+    where q = X // L, r = X % L, and c is the one-period count.
+    """
+    profile = periodic_congruence_union_profile(source)
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/operations.py]: Fixed on the current branch. Scalar prefix admission uses the counting execution plan directly and no longer requires materializing the full periodic profile; large-period complemented cases are covered by regression tests.

File: src/jacobian/math/number_theory/periodic_prefix_count/operations.py [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+        require_admitted_periodic_source(source)
+        period = common_period(source)
+    except ValueError as exc:
+        _reject("execution_bound", str(exc))
+    states, merges = _intersection_bounds(source)
+    full_union = any(
+        len(subset.residues) == parse_canonical_integer(subset.modulus)
+        for subset in source.subsets
+    )
+    if not full_union and (states > 65_535 or merges > 100_000):
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/operations.py]: Fixed on the current branch. Scalar counting dispatches through the admitted FULL_UNION/PERIOD_LIFT/SPARSE_LIFT/INCLUSION_EXCLUSION plan, so a safe period-lift plan is not rejected by the unrelated inclusion-exclusion merge estimate.

File: src/jacobian/math/number_theory/periodic_prefix_count/operations.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        _reject("invalid_source", "source must be a canonical periodic union")
+    if isinstance(cutoff, bool) or not isinstance(cutoff, int):
+        _reject("invalid_cutoff", "cutoff must be a strict integer")
+    if cutoff < 0:
+        _reject("negative_cutoff", "cutoff must be nonnegative")
+    if cutoff > 10**MAX_PERIODIC_INTEGER_DIGITS - 1:
+        _reject(
+            "cutoff_digit_bound",
+            f"cutoff must have at most {MAX_PERIODIC_INTEGER_DIGITS} digits",
+        )
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/periodic_prefix_count/operations.py]: Fixed on the current branch. Prefix cutoff admission has its own 256-digit scalar envelope and output-byte check, independent of the period/profile row bound; a 257-digit cutoff is rejected only at the explicit scalar representation boundary while 256-digit compact counts remain supported.
